src/wp-includes/class-wp-query.php

```diff
@@ -3104,17 +3111,38 @@ public function get_posts() {
 		";
 
 		$this->request = $old_request;
+		$this->count_request = "
+			SELECT COUNT(DISTINCT {$count_field})
+			FROM {$wpdb->posts} $join
+			WHERE 1=1 $where
+		";
 
 		if ( ! $q['suppress_filters'] ) {
 			/**
 			 * Filters the completed SQL query before sending.
 			 *
 			 * @since 2.0.0
+			 * @since x.x.x This query no longer contains a `SQL_CALC_FOUND_ROWS` modifier by default.
 			 *
 			 * @param string   $request The complete SQL query.
 			 * @param WP_Query $query   The WP_Query instance (passed by reference).
 			 */
 			$this->request = apply_filters_ref_array( 'posts_request', array( $this->request, &$this ) );
+
+			if ( is_string( $this->request ) && str_contains( $this->request, 'SQL_CALC_FOUND_ROWS' ) ) {
+				_deprecated_argument(
+					'The posts_request filter',
+					'x.x.x',
+					sprintf(
+						/* translators: 1: SQL query modifier 2: SQL query */
+						__( 'The %1$s query modifier should no longer be added to queries because results are no longer counted with %2$s by default.' ),
+						'<code>SQL_CALC_FOUND_ROWS</code>',
+						'<code>SELECT FOUND_ROWS()</code>'
+					)
+				);
+
+				$this->count_request = 'SELECT FOUND_ROWS()';
+			}
```

Comment:
As we are checking both `posts_request_ids` and `posts_request` filters, let's make this generic, reusable and make it into it's own method. This will also make it more testable.